Assets/Scripts/Framework/Statemachine/BaseState.cs

```diff
@@ -0,0 +1,14 @@
+using UnityEngine;
+
+public abstract class BaseState : MonoBehaviour
+{
+    protected StateMachine Parent;
+    public bool IsValidToSwitch { get; protected set; }
+
+    public void SetParent(StateMachine targetMachine) => Parent = targetMachine;
+
+    public virtual void EnterState(StateMachine stateMachine) => IsValidToSwitch = false;
```

Comment:
Sure you want to make EnterState virtual? Seems like it's a rather important method for it to be optional.

Reply:
make it abstracts? @BNS-MarkUlrich

Reply:
Dan mis je wat er gebeurd in de body.

Reply:
Yes. Seeing as all the other methods like it are, I would say it makes sense as EnterState and ExitState are probably the most important methods of BaseState and any inheritors absolutely cannot forget to implement it or by accident create something to bypass EnterState, wouldn't want that!

Reply:
> Dan mis je wat er gebeurd in de body.

I'm not sure you should want a check like that though. If you do that only really means you've created an edge case that should be avoidable